src/lib/pickles/plonk_checks/scalars.ml

```diff
@@ -3303,6 +4746,324 @@ module Tock : S = struct
               * cell (var (Witness 3, Curr))
               * cell (var (Witness 4, Curr))
             + cell (var (Coefficient 9, Curr)) ) )
+    + if_feature
```

Comment:
All of this piece of code seems very very similar (or even the same) as the one in the top rows of this same file... Shouldn't we factor this out?